### DISCUSSION TITLE
Add this_plugin_version_check_for_updates to core?

### DISCUSSION THREAD

**Discussion by scottcwilson** [zencart/zencart#5785] [Ideas]
Some plugins such as  Square Webpay 
https://www.zen-cart.com/downloads.php?do=file&id=2345 
and Colorbox for Zen Cart 
https://www.zen-cart.com/downloads.php?do=file&id=1322

provide per-plugin config settings to disable phone-home version checking.  They use a function called this_plugin_version_check_for_updates.  I wonder if this should be added to the core so that each plugin that uses it doesn't have to include it?   (currently, plugins do a `function_exists` check and define it if it doesn't exist.)

**Comment by scottcwilson** [comment]
Docs:

https://docs.zen-cart.com/dev/plugins/tips/#automatic-new-version-checks
https://docs.zen-cart.com/user/plugins/version_check/

**Comment by lat9** [comment]
Looks like a good idea, although the function would need to 'adapt' to use the VersionServer class as in the `plugin_support.php` functions' `plugin_version_check_for_updates` function.

**Comment by drbyte** [comment]
Forgive me for not downloading those plugins to see what they've done to re-implement something in core for another purpose.

What's the code do? 
What's the point?
